mkosi/run.py

```diff
@@ -0,0 +1,273 @@
+import contextlib
+import ctypes
+import ctypes.util
+import multiprocessing
+import os
+import pwd
+import shlex
+import signal
+import subprocess
+import sys
+from pathlib import Path
+from types import FrameType
+from typing import Any, Iterable, Iterator, Mapping, Optional, Sequence
+
+from mkosi.backend import MkosiState
+from mkosi.log import ARG_DEBUG, MkosiPrinter, die
+from mkosi.types import _FILE, CompletedProcess, PathString, Popen
+
+CLONE_NEWNS = 0x00020000
+CLONE_NEWUSER = 0x10000000
+
+
+def unshare(flags: int) -> None:
+    libc_name = ctypes.util.find_library("c")
+    if libc_name is None:
+        die("Could not find libc")
+    libc = ctypes.CDLL(libc_name, use_errno=True)
+
+    if libc.unshare(ctypes.c_int(flags)) != 0:
+        e = ctypes.get_errno()
+        raise OSError(e, os.strerror(e))
+
+
+def read_subrange(path: Path) -> tuple[int, int]:
+    for line in path.read_text().splitlines():
+        name, start, count = line.split(":")
+
+        if name == pwd.getpwuid(os.getuid()).pw_name:
```

Comment:
That's maybe an esoteric worry, but if the file contains UIDs that are not mapped on the host (where we run `pwd.getpwuid`) this will throw a `KeyError`.